src/encoding.md

```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-serialize-csv]: #ex-serialize-csv
+<a name="ex-serialize-csv"></a>
+## Serialize records to CSV
+
+[![csv-badge]][csv] [![cat-encoding-badge]][cat-encoding]
+
+CSV writer supports automatic serialization from Rust types into CSV
+records. This example shows how to serialize a Rust tuple.
+
+```rust
+# #[macro_use]
+#extern crate error_chain;
+extern crate csv;
+
+use std::io;
+#
+# error_chain! {
+#     foreign_links {
+#         CSVError(csv::Error);
+#         IOError(std::io::Error);
+#    }
+# }
+
+fn run() -> Result<()> {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
+
+    wtr.write_record(&["Name", "Place", "ID"])?;
+
+    wtr.serialize(("Mark", "Sydney", 87))?;
+    wtr.serialize(("Ashley", "Dublin", 32))?;
+    wtr.serialize(("Akshat", "Delhi", 11))?;
+
+    wtr.flush()?;
+    Ok(())
+}
+#
+# quick_main!(run);
+```
+
+[ex-csv-serde]: #ex-csv-serde
+<a name="ex-csv-serde"></a>
+## Serialize records to CSV using Serde
+
+[![csv-badge]][csv] [![serde-badge]][serde] [![cat-encoding-badge]][cat-encoding]
+
+The following example shows how to serialize custom structs as CSV records. For this,
+make use of the [serde] crate.
+
+```rust
+# #[macro_use]
+#extern crate error_chain;
+extern crate csv;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+
+use std::io;
+#
+# error_chain! {
+#    foreign_links {
+#        IOError(std::io::Error);
+#        CSVError(csv::Error);
+#    }
+# }
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
+struct Record<'a> {
+    name: &'a str,
+    place: &'a str,
+    id: u64,
+}
+
+fn run() -> Result<()> {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
```

Comment:
i would mention the key identifiers (not described in previous exame) in description linking to the docs